Extensions/Effects/JsExtension.js

```diff
@@ -138,12 +138,115 @@ module.exports = {
       .setDescription(_('Alter the colors to sepia'))
       .addIncludeFile('Extensions/Effects/sepia-pixi-filter.js');
     const sepiaProperties = sepiaEffect.getProperties();
-    sepiaProperties.set(
-      'opacity',
-      new gd.PropertyDescriptor(/* defaultValue= */ '1')
-        .setLabel(_('Opacity (between 0 and 1)'))
-        .setType('number')
-    );
+    sepiaProperties.set('opacity', new gd.PropertyDescriptor(/* defaultValue= */ '1').setLabel(_('Opacity (between 0 and 1)')).setType('number'));
+
+    const setFilterProperties = (propertiesSchema, filterProperties) => {
+      Object.keys(propertiesSchema).forEach(property => {
+        const value = propertiesSchema[property];
+        filterProperties.set(property, new gd.PropertyDescriptor(/* defaultValue= */ value.default).setLabel(value.name + ' (' + value.description + ')').setType('number'));
```

Comment:
I know that you want to avoid putting ' (' + ... + ')' everywhere, but I was thinking of putting the name of the parameter (value.name) in the UI in the IDE 😬😬

I know that you want to save a few lines, but I would really prefer to stick to the API without introducing now another layer of abstraction which will make newcomers wonder what is this thing that differs from the documentation.

If there are improvements to the UI, let's make them in the IDE rather than baking them in the extensions :)

Reply:
removed the utility approach and now using the standard way